### PR TITLE
Ansible Security Hardening for:

### PR DESCRIPTION
**Pull request by satya-rajesh**
1: Cron
2: SSH

### CODE REVIEW

File: examples/Dockerfile [hunk 1]
```diff
@@ -20,6 +20,8 @@ shadow \
 procps-ng \
 awk \
 iptables \
+cronie \
+openssh \
```
Comment (satya-rajesh) [inline examples/Dockerfile:24]: Please include in ansible-security-hardening/roles/cis/tasks/main.yml as your changes are not currently running in build: https://travis-ci.org/vmware/ansible-security-hardening/builds/636311091?utm_source=github_status&utm_medium=notification

Please also add these packages in README.md:

### Prerequisites

* ansible (host)
* python3 (target)
* procps-ng (target)
* awk (target)
* iptables (target)
---here--

#install python3 for ansible
#install shadow for cis rule 6.1.2
#install awk for cis rule 6.1.10
#install procps-ng for cis rule 3.1.1
#install iptables for cis rule 3.6.2
#install photon-release for cis rule 1.7.1.6
tdnf --installroot ~/testroot \
--releasever 3.0 --nogpgcheck \
install python3 shadow awk procps-ng iptables photon-release -y
```
----also above---

File: roles/cis/tasks/5/5.1.3.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+- name: "5.1.3 Ensure permissions on /etc/cron.hourly are configured (Scored)"
+  file:
+    path: "/etc/cron.hourly"
+    owner: root
+    group: root
+    mode: 0700
```
Comment (satya-rajesh) [inline roles/cis/tasks/5/5.1.3.yml]: Is this 600 ? The spec says 600, please check

File: roles/cis/tasks/5/5.1.4.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+- name: "5.1.4 Ensure permissions on /etc/cron.daily are configured (Scored)"
+  file:
+    path: "/etc/cron.daily"
+    owner: root
+    group: root
+    mode: 0700
```
Comment (satya-rajesh) [inline roles/cis/tasks/5/5.1.4.yml]: 600 here.

File: roles/cis/tasks/5/5.1.5.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+---
+- name: "5.1.5 Ensure permissions on /etc/cron.weekly are configured (Scored)"
+  file:
+    path: "/etc/cron.weekly"
+    owner: root
+    group: root
+    mode: 0700
```
Comment (satya-rajesh) [inline roles/cis/tasks/5/5.1.5.yml]: 600 here aswell

File: roles/cis/tasks/5/5.1.6.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+---
+- name: "5.1.6 Ensure permissions on /etc/cron.monthly are configured (Scored)"
+  file:
+    path: "/etc/cron.monthly"
+    owner: root
+    group: root
+    mode: 0700
```
Comment (satya-rajesh) [inline roles/cis/tasks/5/5.1.6.yml]: 600 here aswell

File: roles/cis/tasks/5/5.1.7.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+---
+- name: "5.1.7 Ensure permissions on /etc/cron.d are configured (Scored)"
+  file:
+    path: "/etc/cron.d"
+    owner: root
+    group: root
+    mode: 0700
```
Comment (satya-rajesh) [inline roles/cis/tasks/5/5.1.7.yml]: 600 here

File: roles/cis/tasks/5/5.2.2.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+---
+- name: "5.2.2 Ensure SSH Protocol is set to 2 (Scored)"
+  vars:
+    file: /etc/ssh/sshd_config
+  lineinfile:
+    path: "{{ file }}"
+    line: "Protocol 2"
+
+#- name: "5.2.2 - restart sshd"
```
Comment (satya-rajesh) [inline roles/cis/tasks/5/5.2.2.yml]: PLease remove the commented code

File: roles/cis/tasks/5/verify/5.1.3.yml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+        path: /etc/cron.hourly
+      register: etc_cron_hourly_result
+    - fail:
+        msg: "5.1.3 failed to verify uid. expected: 0. found: {{ etc_cron_hourly_result.stat.uid }}"
+      when: etc_cron_hourly_result.stat.uid != 0
+    - fail:
+        msg: "5.1.3 failed to verify gid. expected: 0. found: {{ etc_cron_hourly_result.stat.gid }}"
+      when: etc_cron_hourly_result.stat.gid != 0
+    - fail:
+        msg: "5.1.3 failed to verify permissions. expected: 0700. found: {{ etc_cron_hourly_result.stat.mode }}"
```
Comment (satya-rajesh) [inline roles/cis/tasks/5/verify/5.1.3.yml]: test for 600

File: roles/cis/tasks/5/verify/5.1.4.yml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+        path: /etc/cron.daily
+      register: etc_cron_daily_result
+    - fail:
+        msg: "5.1.4 failed to verify uid. expected: 0. found: {{ etc_cron_daily_result.stat.uid }}"
+      when: etc_cron_daily_result.stat.uid != 0
+    - fail:
+        msg: "5.1.4 failed to verify gid. expected: 0. found: {{ etc_cron_daily_result.stat.gid }}"
+      when: etc_cron_daily_result.stat.gid != 0
+    - fail:
+        msg: "5.1.4 failed to verify permissions. expected: 0700. found: {{ etc_cron_daily_result.stat.mode }}"
```
Comment (satya-rajesh) [inline roles/cis/tasks/5/verify/5.1.4.yml]: test for 600

File: roles/cis/tasks/5/verify/5.1.5.yml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+      register: etc_cron_weekly_result
+    - fail:
+        msg: "5.1.5 failed to verify uid. expected: 0. found: {{ etc_cron_weekly_result.stat.uid }}"
+      when: etc_cron_weekly_result.stat.uid != 0
+    - fail:
+        msg: "5.1.5 failed to verify gid. expected: 0. found: {{ etc_cron_weekly_result.stat.gid }}"
+      when: etc_cron_weekly_result.stat.gid != 0
+    - fail:
+        msg: "5.1.5 failed to verify permissions. expected: 0700. found: {{ etc_cron_weekly_result.stat.mode }}"
+      when: etc_cron_weekly_result.stat.mode != "0700"
```
Comment (satya-rajesh) [inline roles/cis/tasks/5/verify/5.1.5.yml]: test for 600

File: roles/cis/tasks/5/verify/5.1.6.yml [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+        path: /etc/cron.monthly
+      register: etc_cron_monthly_result
+    - fail:
+        msg: "5.1.6 failed to verify uid. expected: 0. found: {{ etc_cron_monthly_result.stat.uid }}"
+      when: etc_cron_monthly_result.stat.uid != 0
+    - fail:
+        msg: "5.1.6 failed to verify gid. expected: 0. found: {{ etc_cron_monthly_result.stat.gid }}"
+      when: etc_cron_monthly_result.stat.gid != 0
+    - fail:
+        msg: "5.1.6 failed to verify permissions. expected: 0700. found: {{ etc_cron_monthly_result.stat.mode }}"
```
Comment (satya-rajesh) [inline roles/cis/tasks/5/verify/5.1.6.yml]: 600 here

File: roles/cis/tasks/5/verify/5.1.7.yml [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+      register: etc_cron_d_result
+    - fail:
+        msg: "5.1.7 failed to verify uid. expected: 0. found: {{ etc_cron_d_result.stat.uid }}"
+      when: etc_cron_d_result.stat.uid != 0
+    - fail:
+        msg: "5.1.7 failed to verify gid. expected: 0. found: {{ etc_cron_d_result.stat.gid }}"
+      when: etc_cron_d_result.stat.gid != 0
+    - fail:
+        msg: "5.1.7 failed to verify permissions. expected: 0700. found: {{ etc_cron_d_result.stat.mode }}"
+      when: etc_cron_d_result.stat.mode != "0700"
```
Comment (satya-rajesh) [inline roles/cis/tasks/5/verify/5.1.7.yml]: test for 600

File: examples/Dockerfile (hunk 1 above)
Comment (shreyasbe) [reply examples/Dockerfile:24]: Adressed

Comment (shreyasbe) [reply examples/Dockerfile:24]: Addressed

**Review by ppadmavilasom — APPROVED** [review]

**Comment by ppadmavilasom** [comment]
please address comments and merge

File: roles/cis/tasks/5/verify/5.1.7.yml (hunk 12 above)
Comment (shreyasbe) [reply roles/cis/tasks/5/verify/5.1.7.yml]: Changes done.

File: roles/cis/tasks/5/verify/5.1.6.yml (hunk 11 above)
Comment (shreyasbe) [reply roles/cis/tasks/5/verify/5.1.6.yml]: Changes done.

File: roles/cis/tasks/5/verify/5.1.5.yml (hunk 10 above)
Comment (shreyasbe) [reply roles/cis/tasks/5/verify/5.1.5.yml]: Changes done.

File: roles/cis/tasks/5/verify/5.1.4.yml (hunk 9 above)
Comment (shreyasbe) [reply roles/cis/tasks/5/verify/5.1.4.yml]: Changes done.

File: roles/cis/tasks/5/verify/5.1.3.yml (hunk 8 above)
Comment (shreyasbe) [reply roles/cis/tasks/5/verify/5.1.3.yml]: Changes done.

File: roles/cis/tasks/5/5.1.7.yml (hunk 6 above)
Comment (shreyasbe) [reply roles/cis/tasks/5/5.1.7.yml]: Changes done.

File: roles/cis/tasks/5/5.2.2.yml (hunk 7 above)
Comment (shreyasbe) [reply roles/cis/tasks/5/5.2.2.yml]: Changes done.

File: roles/cis/tasks/5/5.1.5.yml (hunk 4 above)
Comment (shreyasbe) [reply roles/cis/tasks/5/5.1.5.yml]: Changes done.

File: roles/cis/tasks/5/5.1.4.yml (hunk 3 above)
Comment (shreyasbe) [reply roles/cis/tasks/5/5.1.4.yml]: Changes done.

File: roles/cis/tasks/5/5.1.3.yml (hunk 2 above)
Comment (shreyasbe) [reply roles/cis/tasks/5/5.1.3.yml]: Changes done.

**Comment by satya-rajesh** [comment]
There are couple of warnings observed here:
https://travis-ci.org/vmware/ansible-security-hardening/builds/637520730?utm_source=github_status&utm_medium=notification

could you please fix them.

TASK [cis : 5.2.11 Ensure only approved ciphers are used (Scored)] *************
changed: [/root/testroot]
 [WARNING]: The loop variable 'item' is already in use. You should set the
`loop_var` value in the `loop_control` option for the task to something else to
avoid variable collisions and unexpected behavior.
TASK [cis : 5.2.12 Ensure only approved MAC algorithms are used (Scored)] ******
changed: [/root/testroot] => (item={u'line': u'# MAC algorithms'})
changed: [/root/testroot] => (item={u'line': u'MACs hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-512,hmac-sha2-256,umac-128@openssh.com'})
 [WARNING]: The loop variable 'item' is already in use. You should set the
`loop_var` value in the `loop_control` option for the task to something else to
avoid variable collisions and unexpected behavior.

**Comment by satya-rajesh** [comment]
Looks good, approved.

**Pull request merged**